forte/data/data_store.py

```diff
@@ -198,14 +202,161 @@ def __init__(
+                # fields. `diff` records fields in the current class but not
+                # in the serialized objects.
+                for t, v in cls._type_attributes.items():
+                    change_map = {}
+                    contradict_loc = []
+                    diff = set(v["attributes"].items()) - set(
+                        store._type_attributes[t]["attributes"].items()
+                    )
+                    for f in diff:
+                        # if only orders are different, switch order
+                        if f[0] in store._type_attributes[t]["attributes"]:
+                            change_map[f[1]] = store._type_attributes[t][
+                                "attributes"
+                            ][f[0]]
+                        # if fields contradict, fill them with None
+                        else:
+                            contradict_loc.append(f[1])
+
+                    if len(change_map) > 0:
+                        if not silent_mode:
+                            logger.warning(
+                                "Saved %s objects have %s different orders of "
+                                "attribute fields to the current datastore. "
+                                "They are reordered to match the fields in "
+                                "the current datastore class.",
+                                t,
+                                len(change_map),
+                            )
+                        # change the data in lists in the `__elements` list
+                        for d in store._DataStore__elements[t]:
+                            d[:] = [
+                                d[change_map[i]] if i in change_map else d[i]
+                                # throw fields that are redundant
+                                for i in range(
+                                    max(v["attributes"].values()) + 1
+                                )
+                            ]
+                    if len(contradict_loc) > 0:
+                        if not silent_mode:
+                            logger.warning(
+                                "Saved %s objects have %s attribute fields "
+                                "that could not be identified. "
+                                "These fields are filled with `None`. This may "
+                                "due to user's modifications of fields.",
+                                t,
+                                len(contradict_loc),
+                            )
+                        if accept_none:
+                            for d in store._DataStore__elements[t]:
+                                for idx in contradict_loc:
+                                    d[idx] = None
+                        else:
+                            raise ValueError(
+                                "Saved objects have unidentified fields, which"
+                                " raise an error."
```

Comment:
We should spell out the unidentified class, field and value to help debugging

Reply:
fixed